build-scripts/createRelease.ts

```diff
@@ -0,0 +1,62 @@
+/*!
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as child_process from 'child_process'
+import * as fs from 'fs-extra'
+import * as path from 'path'
+
+const changesDirectory = '.changes'
+const nextReleaseDirectory = path.join(changesDirectory, 'next-release')
+// tslint:disable-next-line:no-var-requires no-unsafe-any
+const releaseVersion = require(path.join('..', 'package.json')).version
+const changesFile = path.join(changesDirectory, `${releaseVersion}.json`)
+
+fs.mkdirpSync(nextReleaseDirectory)
+
+const changeFiles = fs.readdirSync(nextReleaseDirectory)
+if (changeFiles.length === 0) {
+    console.log('Error! no changes to release!')
+    process.exit(-1)
+}
+try {
+    fs.accessSync(changesFile)
+    console.log(`Error! changelog file ${changesFile} already exists for version ${releaseVersion}!`)
+    process.exit(-1)
+} catch (err) {
+    // This is what we want to happen, the file should not exist
+}
+
+const now = new Date()
+const timestamp = `${now.getFullYear()}-${now.getMonth()}-${now.getDay()}`
+const changelog: any = {
+    date: timestamp,
+    version: releaseVersion,
+    entries: []
+}
+
+for (const changeFile of changeFiles) {
+    const file = JSON.parse(fs.readFileSync(path.join(nextReleaseDirectory, changeFile)).toString())
+    // tslint:disable-next-line: no-unsafe-any
+    changelog.entries.push(file)
+}
+
+// Write changelog file
+fs.writeFileSync(changesFile, JSON.stringify(changelog, undefined, '\t'))
+const fileData = fs.readFileSync('CHANGELOG.md')
+let append = `## ${releaseVersion} ${timestamp}\n\n`
+for (const changeFile of changeFiles) {
+    const file: any = JSON.parse(fs.readFileSync(path.join(nextReleaseDirectory, changeFile)).toString())
+    // tslint:disable-next-line: no-unsafe-any
+    append += `- **${file.type}** ${file.description}\n`
```

Comment:
The Jetbrains changelog groups by type. Could this script do this as well without relying on the filenames to provide grouping as a side effect?

Reply:
Added a sort

Reply:
If I understand correctly, the sort is using the filenames, which is not ideal to rely on. Prefer to sort `changelog.entries` on `type`, since this is the more concrete contract of the data.

Reply:
sure sorted by type